components/Admin/Admin.js

```diff
@@ -57,12 +62,22 @@ class Admin extends React.Component {
     })
   }
 
+  renderEditPage = (location) => {
+    this.setState(Object.assign({}, this.state, {showEditPage: true, currentLocation: location}))
```

Comment:
I was wrong about this.  `setState` does a shallow merge so you don't need this crappy `Object.assign` foo. 

```js
  this.setState({ showEdidtPage: true, currentLocation: location })
```